src/scipp/spatial/transform/translation.py

```diff
@@ -0,0 +1,30 @@
+from typing import Union, Sequence
+import numpy as _np
+
+from ..._scipp import core as _core_cpp
+
+
+def from_vector(*,
```

Comment:
I feel that having multiple `from_vector(s)` functions (e.g., for scaling and translation) is too confusing.

Furthermore, I am not sure nesting the submodules so much is helpful? `scipp.spatial.transform.rotation` etc.? While this is clearly trying to mimic some of the Scipy structure, it actually isn't:

- In scipy `Rotation` is a class, not a module. We have no use for this class, so making a module with the same name is maybe more confusing than helpful to the user?
- In scipy there is no `translation` or `scaling` class or submodule within `transform`.

So I am thinking that we should consider simplifying the submodule structure?

For example, instead of
```python
scipp.spatial.transform.rotation.from_rotvec
```
other options could be
```python
scipp.spatial.rotation.from_rotvec
scipp.spatial.rotation_from_rotvec
```
But maybe we should think more outside the box, and follow the same pattern as in the "normal" creation functions. Consider
```python
sc.array(dims=[...], values=a, dtype='float64')
```
In the above, the element dtype is decoupled from the function name. Maybe we can do the same for spatial transforms?
```python
from_rotvec(value=vec, dtype='rotation')
from_rotvec(value=vec, dtype='linear_transform')
```
I am not sure. Maybe we should just discuss this in a call today.